### PR TITLE
Run with existing matrices

### PR DESCRIPTION
**Pull request by assafmu**
Implementation of issue 139. Also moved ...hakab_data, so it's easier to run from terminals without PYTHON_PATH trickery.
{closing #139}

### CODE REVIEW

**Comment by bentheiii** [comment]
the whole point of the __data__ is that the setupscripts can run without habving the dependencies installed

File: src/corona_hakab_model/main.py [hunk 1]
```diff
@@ -48,18 +69,24 @@ def main():
...
+        population_data = PopulationData.import_population_data(args.population_data)
 
-    if args.sub_command == "generate":
-        with open(args.output, "wb") as w:
-            write(gm.matrix_data.matrix, w)
-        return
+    else:
+        gm = GenerationManger(circles_consts=circles_consts, matrix_consts=matrix_consts)
+        if args.sub_command == 'generate':
+            gm.matrix_data.export('matrix')
```
Comment (tal-nc) [inline src/corona_hakab_model/main.py:79]: this is exported upon generation by default anyway, no?
i think you would export the matrix twice this way.

File: src/corona_hakab_model/generation/matrix_generator.py [hunk 2]
```diff
@@ -28,13 +31,23 @@ def __init__(self):
         self.matrix = None
 
     # todo make this work, using the parasymbolic matrix serialization.
-    def export(self, export_path, file_name: str):
-        pass
+    def export(self, export_path: str):
+        with open(export_path+'.'+self.matrix_type, 'wb') as f:
+            bsa.universal.write(self.matrix, f)
```
Comment (tal-nc) [inline src/corona_hakab_model/generation/matrix_generator.py:36]: have you tested this to make sure that it works on the parasymbolic matrix?

File: src/corona_hakab_model/main.py (hunk 1 above)
Comment (assafmu) [reply src/corona_hakab_model/main.py:79]: No enabled by default, but I should still refactor it.

File: src/corona_hakab_model/generation/matrix_generator.py (hunk 2 above)
Comment (assafmu) [reply src/corona_hakab_model/generation/matrix_generator.py:36]: Yes, I've checked it on the parasymbolic matrix type.

**Comment by assafmu** [comment]
> the whole point of the **data** is that the setupscripts can run without habving the dependencies installed

Should I instead separate the data creation part to a separate script, and have main.py assume that the files already exist?

**Comment by bentheiii** [comment]
`__data__` only stores metadata information about the project

**Comment by assafmu** [comment]
__data__ moved back to original place, cmd works with Ben's PYTHONPATH batch.
And data exporting is done only explicitly (and not on init), so now it happens once.

File: src/corona_hakab_model/main.py [hunk 3]
```diff
@@ -3,41 +3,78 @@
...
+    parser.add_argument('--seed',
+                        dest='seed',
+                        type=int,
+                        default=None,
+                        help='Set the random seed. Use only for exact reproducibility. By default, generate new seed.')
     parser.add_argument("--version", action="version", version=__version__)
+
     args = parser.parse_args()
+    set_seeds(args.seed)
+    if args.sub_command == 'generate':
```
Comment (RoyIsland) [inline src/corona_hakab_model/main.py:70]: יכול להיות נחמד שתהיה גם אופציה להריץ את הגינרוט והסימולציה בבת אחת

File: src/corona_hakab_model/generation/circles_generator.py [hunk 4]
```diff
@@ -172,14 +171,16 @@ def fill_social_circles_by_agent_index(self):
...
+    def _fill_population_data(self):
         # fill population data with my data
         self.population_data.agents = self.agents
         self.population_data.social_circles_by_connection_type = self.social_circles_by_connection_type
         self.population_data.geographic_circles = self.geographic_circles
         self.population_data.geographic_circle_by_agent_index = self.geographic_circle_by_agent_index
         self.population_data.social_circles_by_agent_index = self.social_circles_by_agent_index
 
+    def export(self):
+        self._fill_population_data()
```
Comment (RoyIsland) [inline src/corona_hakab_model/generation/circles_generator.py:183]: אתה קורא לפעולה הזאת כבר ביצירה של CirclesGenerator (שורה 108)

Comment (assafmu) [reply src/corona_hakab_model/generation/circles_generator.py]: צודק. מחקתי

File: src/corona_hakab_model/main.py (hunk 3 above)
Comment (assafmu) [reply src/corona_hakab_model/main.py:72]: הוספתי את האפשרות להריץ 
main.py all
ודרכו להריץ את שני האחרים, ועוד שנוסיף בהמשך

File: src/corona_hakab_model/generation/matrix_generator.py [hunk 5]
```diff
@@ -308,6 +321,9 @@ def _add_small_circle_connections(self, circle: SocialCircle, connections: List[
 
             agent_id_pool.difference_update(to_remove)
 
+    def export_matrix_data(self,export_dir='..\..\output',export_filename='matrix.bsa'):
```
Comment (RoyIsland) [inline src/corona_hakab_model/generation/matrix_generator.py:324]: no use

**Pull request merged**